### ISSUE TITLE
Improve figwheel-bridge.js: do not manually create document and localStorage

### ISSUE THREAD

**Issue by drapanjanas** [drapanjanas/re-natal#28]
Problems with Firebase related to this were reported on slack channel. There might more libs which might be 'fooled' by existing `window.document` object and work not properly in RN environment.

The solution is to make Figwheel 'think' that it is Node.js environment rather than browser.

**Comment by seantempesta** [comment]
Copying what @artemyarulin wrote:


artemyarulin [2:16 PM] 
@seantempesta: Well, I’m using different approach. In figwheel source:
```
(defn get-websocket-imp []
  (cond
    (utils/html-env?) (aget js/window "WebSocket")
    (utils/node-env?) (try (js/require "ws")
                           (catch js/Error e
                             nil))
    :else nil))
(defn html-env? [] (goog/inHtmlDocument_))
```
Meaning Figwheel will use Browser WS implementation and in order to make it happen figwheel-react-native creates all those window.document and so on (actually cannot find how window.WebSocker got created in figwheel-react-native, dunno...), which in turn will make thinking Figwheel that it’s in a browser environment, while RN actually more close to NodeJS, although `require 'ws'` wouldn’t work with RN, unless we hack it like that:
``` 
;; Let's tell figwheel to get WS lib from the right place
(when (re-find #"figwheel.client.socket.js" url)
  (aset (aget js/figwheel "client" "socket") "get_websocket_imp" (fn[](js/require "WebSocket"))))
```
After that everything works like it should be. Figwheel don’t see document/window, meaning it will behave like under NodeJS environment, while RN implementation of WS would be used

**Comment by artemyarulin** [comment]
Thanks, full source is there https://github.com/artemyarulin/ktoa/blob/master/src/ktoa/repl.cljc

**Comment by drapanjanas** [comment]
Thanks guys, lately I'm a bit busy witho other exciting development stuff ;) but will try to hack this issue over the weekend in a branch.

**Comment by drapanjanas** [comment]
Should be fixed commit dfa5cdece8e3ff56f37ebb64323ba895a9b07761 (referenced wrong issue in commit message)
